Core/SVS/src/drawer_asio.cpp

```diff
@@ -0,0 +1,186 @@
+#include <iostream>
+#include <cstdlib>
+#include <asio.hpp>
+#include "drawer.h"
+#include "common.h"
+#include "sgnode.h"
+
+/**** 
+ * Note: This is a new implementation of the drawer socket that uses the asio library
+ *          it will be used in the cmake build process, replacing the old platform_specific code 
+ ****/
+
+std::ostream& write_vec3(std::ostream& os, const vec3& v)
+{
+    os << v(0) << " " << v(1) << " " << v(2);
+    return os;
+}
+
+class ipcsocket
+{
+    public:
+        ipcsocket() : socket(io_context), is_connected(false) {}
+
+        ~ipcsocket()
+        {
+            disconnect();
+        }
+
+        bool connect(const std::string& port)
+        {
+            if (is_connected) {
+                disconnect();
+            }
+            try {
+                asio::ip::tcp::resolver resolver(io_context);
+                auto endpoints = resolver.resolve("127.0.0.1", port);
+
+                asio::connect(socket, endpoints);
+                is_connected = true;
+            } catch (std::exception& e) {
+                // ignore
+            }
+            return is_connected;
+        }
+
+        void disconnect()
+        {
+            if (!is_connected) {
+                return;
+            }
+            try {
+                socket.close();
+            } catch (std::exception& e) {
+                // ignore
```

Comment:
I think it would be better to just print something here, in case we run into a strange issue.